src/plugins/lua/LuaEvent.cpp

```diff
@@ -304,13 +337,23 @@ void CALLBACK LuaEventCallback(unsigned int ID, void* pData, BLECHVALUE* pValues
 }
 
 LuaEvent::LuaEvent(std::string_view name, std::string_view expression,
-	const sol::function& func, LuaEventProcessor* processor, Blech& blech)
+	const sol::function& func,
+	const sol::table& options, // Accept options table
+	LuaEventProcessor* processor, Blech& blech)
 	: m_name(name)
 	, m_expression(expression)
 	, m_function(func)
 	, m_processor(processor)
 	, m_blech(blech)
+	, m_keep_links(false)
 {
+	if (options.valid())
+	{
+		if (auto keep_links_opt = options["keep_links"]; keep_links_opt.valid())
+		{
+			m_keep_links = keep_links_opt.get_or(false); // Get the value or default to false
+		}
+	}
```

Comment:
If you pass in the optional table as I described in another comment, you can do something like this. The valid checks are redundant in this case.

```
if (options.has_value())
{
    m_keepLinks = options["keep_links"].get_or_(false);
}
```